search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -95,20 +94,23 @@ class TemporalEntityHandler(
                 .body(BadRequestDataResponse(e.message))
         }
 
-        val temporalEntityAttribute = temporalEntityAttributeService.getForEntity(
+        val temporalEntityAttributes = temporalEntityAttributeService.getForEntity(
             entityId.toUri(),
             temporalQuery.attrs,
             contextLink
-        ).awaitFirstOrNull() ?: throw ResourceNotFoundException(entityNotFoundMessage(entityId))
+        ).collectList().awaitFirst().ifEmpty { throw ResourceNotFoundException(entityNotFoundMessage(entityId)) }
 
-        val results = attributeInstanceService.search(temporalQuery, temporalEntityAttribute).awaitFirst()
+        val attributeAndResultsMap = temporalEntityAttributes.map {
+            it to attributeInstanceService.search(temporalQuery, it).awaitFirst()
+        }.toMap()
 
-        val jsonLdEntity = loadEntityPayload(temporalEntityAttribute, bearerToken).awaitFirst()
+        val jsonLdEntity = loadEntityPayload(attributeAndResultsMap.keys.first(), bearerToken).awaitFirst()
         val jsonLdEntityWithTemporalValues = temporalEntityAttributeService.injectTemporalValues(
             jsonLdEntity,
-            listOf(results),
+            attributeAndResultsMap.values.toList(),
```

Comment:
it could have been improved by taking rid of some of the Reactor legacy but it seems to work as expected

Reply:
You 're talking about loadEntityPayload() ? Last time I tried I got some timeouts. I'm not mastering reactor yet

Reply:
No, I'm talking about the whole selected block that reflects some past workarounds due to Reactor constraints that could be removed now. Let's talk about it tomorrow.